### PR TITLE
WAZO-2291: add recorded calls filter

### PR DESCRIPTION
**Pull request by bloom1**

### CODE REVIEW

File: wazo_call_logd/database/queries/call_log.py [hunk 1]
```diff
@@ -178,6 +178,12 @@ def _apply_filters(self, query, params):
         if params.get('start_id'):
             query = query.filter(CallLog.id >= params['start_id'])
 
+        if params.get('recorded'):
+            if params['recorded'] is True:
+                query = query.filter(CallLog.recordings != None)  # noqa
+            elif params['recorded'] is False:
+                query = query.filter(CallLog.recordings == None)  # noqa
```
Comment (fblackburn1) [inline wazo_call_logd/database/queries/call_log.py:185]: FYI you can avoid linter warning with the following syntax
```suggestion
            if params['recorded'] is True:
                query = query.filter(CallLog.recordings.isnot(None))  # noqa
            elif params['recorded'] is False:
                query = query.filter(CallLog.recordings.is(None))  # noqa
```

File: integration_tests/suite/test_cdr.py [hunk 2]
```diff
@@ -1214,6 +1287,68 @@ def test_list_my_cdr_default_limit(self):
...
+        results = self.call_logd.cdr.list_from_user(recorded=True)
+        assert_that(
+            results,
+            has_entries(
+                filtered=0,
+                total=2,
+                items=empty(),
+            ),
+        )
+
```
Comment (fblackburn1) [inline integration_tests/suite/test_cdr.py:1351]: see previous comment

File: integration_tests/suite/test_cdr.py [hunk 3]
```diff
@@ -1068,6 +1076,42 @@ def test_given_call_logs_when_list_cdr_with_from_id_then_list_matching_cdr(self)
...
+        results = self.call_logd.cdr.list(recorded=True)
+        assert_that(
+            results,
+            has_entries(
+                filtered=0,
+                total=2,
+                items=empty(),
+            ),
+        )
+
```
Comment (fblackburn1) [inline integration_tests/suite/test_cdr.py:1114]: No objection to keep it
IMO: this test is already done in the previous test, any other corner case should be done in more closer test (dao)

**Review by fblackburn1 — CHANGES_REQUESTED** [review]
dao test are missing and the case where `recorded=False` is not tested

File: wazo_call_logd/database/queries/call_log.py (hunk 1 above)
Comment (bloom1) [reply wazo_call_logd/database/queries/call_log.py:185]: That's what I tried at first, but I get a 
```
  File "/opt/venv/lib/python3.7/site-packages/sqlalchemy/sql/operators.py", line 230, in operate
    raise NotImplementedError(str(op))
NotImplementedError: <function isnot at 0x7f11b214ef80>
```
Does it do the same for you?

Comment (fblackburn1) [reply wazo_call_logd/database/queries/call_log.py:185]: Oooh maybe because it's a relationship and not a column ... well too bad :man_shrugging:

Comment (fblackburn1) [reply wazo_call_logd/database/queries/call_log.py:185]: In fact it's really cool that this code work with relationship :smile:

Comment (bloom1) [reply wazo_call_logd/database/queries/call_log.py]: Well, anyway that did not work. I had to use .any() instead :smile:

**Review by fblackburn1 — APPROVED** [review]

**Review by fblackburn1 — DISMISSED** [review]

**Review by fblackburn1 — APPROVED** [review]
